server/aesdsocket.c

```diff
@@ -30,6 +33,248 @@
+
+//timer thread to print timestamps
+void timer_thread (union sigval sigval)
+{
+
+	struct timerthread* timtd = (struct timerthread*) sigval.sival_ptr;
+	time_t rawtime;
+	struct tm *info;
+	char *strtime;
+	size_t strbyte;
+
+	time(&rawtime);
+
+	info = localtime(&rawtime);
+
+	strtime = (char*)malloc(MAXSIZE*sizeof(char));
+		if(strtime == NULL){
+			return;
+		}
+
+	strbyte = strftime(strtime, 80, "timestamp: %Y %b %a %H %M %S%n", info);
+
+		if(strbyte == 0){
+			perror("strftime failure!!!");
+			free(strtime);
+		}
+
+	if(pthread_mutex_lock(&mutex)!=0){
+		perror("mutex lock error");
+		printf("mutex lock error in time thread");
+		free(strtime);
+	}
+
+	int writeret = write(timtd->storefilefd, strtime, strbyte);
+	if(writeret != strbyte){
+		syslog(LOG_ERR, "Timestamp error!!!");
+		free(strtime);
+	}
+
+	if(pthread_mutex_unlock(&mutex)!=0){
+		perror("mutex unlock error");
+		printf("mutex unlock error in time thread");
+		free(strtime);
+	}	
+
+	free(strtime);
+}
+
+//thread functionality that receives, writes, reads and sends one packet 
+void* thread_function(void* thread_param){
+
+	struct params* param_value = (struct params*) thread_param;
+	int recvret=0, writeret=0, readret=0, sendret=0;
+	char buf[MAXSIZE];														    //static buffer to recieve the data into
+	char* newline = NULL; char* newline2 =NULL;
+	char* newptr = NULL; char* newptr2 = NULL;
+	size_t buf2_size=0;
+	int sendsize=0;																//size of each line to be sent
+	off_t ret;
+
+	char *IP = inet_ntoa(connection_addr.sin_addr);
+	syslog(LOG_DEBUG, "Accepted connection from %s\n", IP);
+
+			//clear the buffer for input
+	        memset(buf, '\0', sizeof(buf));
+
+		//buffer to read all the values in and realloc if required
+		char* buf2 = (char*)malloc(MAXSIZE*sizeof(char));
+		if(buf2 == NULL){
+			syslog(LOG_ERR, "read buffer malloc failed!!");
+		}
+
+		int mallocsize = MAXSIZE, reallocsize = MAXSIZE;		//realloc sizes for read and write buffer
+		//char* newptr = NULL; char* newptr2 =NULL;				//newptr checks the realloc return 
+		int setback=0, pos=0, end_of_file=0, nextsize=0;
+		
+		do{
+				//receive the bytes
+				recvret = recv(param_value->per_thread_acceptfd, buf, sizeof(buf), 0);
+				if(recvret == -1){
+					syslog(LOG_ERR, "errno: %s", strerror(errno));
+					return NULL;
+				}
+				else{
+					if((mallocsize-buf2_size) < recvret){
+						mallocsize += recvret;
+						newptr = (char*)realloc(buf2, mallocsize* sizeof(char)); //realloc if the recived bytes are more than the size of buffer
+						if(newptr!= NULL){
+							buf2=newptr;
+						}
+					}
+					memcpy(&buf2[buf2_size], buf, recvret);				//copy contents to another buffer so later can append bigger data files
+				}
+
+			buf2_size+=recvret;
+				newline = strchr(buf, '\n');	//check for newline character
+		}while(newline == NULL); 
+
+		pthread_mutex_lock(&mutex);										//lock the write functionality
+		//write into the file
+        writeret = write(storefd, buf2, buf2_size);
```

Comment:
@Mich2899 watch your spacing here, make sure spacing is setup consistently in your editor to make your code more readable.